### DISCUSSION TITLE
Encoding of Nerd Font characters

### DISCUSSION THREAD

**Discussion by ehawman-rosenberg** [JanDeDobbeleer/oh-my-posh#2381] [General]
@JanDeDobbeleer, you've stated that you wish us to use the `\u` encoding for Nerd Font chars when submitting themes.

Can you elaborate on this a bit? When editing, I really appreciate being able to see the actual characters I'm working with, without needing to reference the [cheat sheet](https://www.nerdfonts.com/cheat-sheet). Also, when converting formats (json to yaml, for example), 99.99% of conversion tools are going to go ahead and encode those characters. Unencoding them is a tedious task.

If the `\u` encoding is a strict requirement, may I request that you modify the `config export` function to detect characters in the [Private Use Areas](https://en.wikipedia.org/wiki/Private_Use_Areas) and export them in the desired format?

I smashed together a little script in Python to accomplish this, but ran into some context problems. Namely, when a string is simple then YAML doesn't require surrounding it in quotes, and linters follow this convention. When running the script on a formatted YAML theme, it can't identify which NF chars are embedded in a quoted string and which are sitting by themselves (meaning I can't just blanket wrap them in quotes)

`wsl: ` becomes `wsl: \ue712` which results in:

![image](https://user-images.githubusercontent.com/81652082/172015115-23fdc2f2-7fc8-49df-a294-0f16f2ebb8de.png) (note the properly rendered Shell, Exit, and Path icons)

I'm not quite sure how `config export` behaves, but you probably wouldn't have to fight as hard to determine context for these. Just wanted to share an issue I bumped into when attempting this. 

<details>
  <summary>Hideous Python code purely for reference</summary> 

```python
filename = "C:\\Users\\ehawman\\Documents\\Programming\\GitForks\\oh-my-posh\\themes\\devious-diamonds.omp.yaml"


def get_uni(char):
    unicode = format(ord(char), "04x")
    return f"\\u{unicode.upper()}"

with open(filename, "r", encoding="utf-8") as f:
    contents = f.read()

result = ""

for char in contents:
    if (
        ("\uE000" <= char <= "\uF8FF")
        or ("\uF0000" <= char <= "\uFFFFD")
        or ("\u100000" <= char <= "\u10FFFD")
    ):
        char = get_uni(char)
    result += char

print(result)
```

**Comment by JanDeDobbeleer** [comment]
You're right it doesn't work for YAML. Does for JSON, the migration/export handles that.
